lib/md051.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+
+"use strict";
+
+const { pathToFileURL } = require("url");
+const fs = require("fs");
+
+const { filterTokens, addError } = require("../helpers");
+
+module.exports = {
+  "names": [ "MD051", "no-dead-relative-links" ],
+  "description": "No dead relative links",
+  "tags": [ "links" ],
+  "function": function MD051(params, onError) {
+    filterTokens(params, "inline", (token) => {
+      token.children.forEach((child) => {
+        const { lineNumber, type, attrs } = child;
+        if (type === "link_open") {
+          attrs.forEach((attr) => {
+            if (attr[0] === "href") {
+              const href = attr[1];
+              const url = new URL(href, pathToFileURL(params.name));
+              url.hash = "";
+              const isRelative = href.startsWith("./") ||
```

Comment:
Relative links could also be without `./`, for example, `README.md` is a valid relative `href` but currently we don't take this into account to stay simple, the problem is that we don't want to consider for example `google.com` as a relative link.

Is this implementation viable? Do you have ideas about this?
I think we could only consider relative links as being `href` starting with `./` or `../`, it could be enough for this rule.